src/mpid/ch4/netmod/ofi/ofi_win.c

```diff
@@ -192,7 +192,7 @@ static int win_allgather(MPIR_Win * win, void *base, int disp_unit)
     /* Don't register MR for NULL buffer, because FI_MR_BASIC mode requires
      * that all registered memory regions must be backed by physical memory
      * pages at the time the registration call is made. */
-    if ((!MPIDI_OFI_ENABLE_MR_PROV_KEY && !MPIDI_OFI_ENABLE_MR_VIRT_ADDRESS) || base) {
+    if (base) {
         MPIDI_OFI_CALL(fi_mr_reg(MPIDI_OFI_global.ctx[0].domain,        /* In:  Domain Object */
```

Comment:
Can we add a note here that it is unclear whether NULL is a valid input when these capabilities are set. Thus, we choose a conservative approach here by disabling fi_mr_reg for all NULL base case?

Reply:
That is a good advice, will do!

Reply:
Updated comments.

Reply:
Looks good to me. I believe this PR is ready to merge.

Reply:
> Looks good to me. I believe this PR is ready to merge.

Sound good. Approve?

Reply:
Done :-)